pymc3/step_methods/hmc/base_hmc.py

```diff
@@ -44,7 +44,7 @@ def __init__(self, vars=None, scaling=None, step_scale=0.25, is_cov=False,
         if isinstance(scaling, dict):
             scaling = guess_scaling(Point(scaling, model=model), model=model, vars=vars)
 
-        n = scaling.shape[0]
+        n = model.dict_to_array(model.test_point).shape[0]
```

Comment:
Agreed, I don't think there is so. Perhaps we should push this into `Model` as a property `ndim`.